servlet/src/main/java/io/undertow/servlet/api/DeploymentInfo.java

```diff
@@ -107,6 +107,7 @@
     private boolean escapeErrorMessage = true;
     private boolean sendCustomReasonPhraseOnError = false;
     private boolean useCachedAuthenticationMechanism = true;
+    private boolean preservePathOnForward = true;
```

Comment:
This should be `false` by default. Because, if this is set to `true` by default, it violates the servlet spec. Please refer to the Servlet API document [HttpServletRequest#getRequestURL](https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html#getRequestURL--) which says:

> If this request has been forwarded using RequestDispatcher.forward(javax.servlet.ServletRequest, javax.servlet.ServletResponse), the server path in the reconstructed URL must reflect the path used to obtain the RequestDispatcher, and not the server path specified by the client.

By this change, the option will be able to be configurable from the servlet extension mechanism. However, I think it's also better to add a new system property as a configurable switch for users who would like to revert the behavior to the previous version of JBoss (JBoss EAP 6.x). Because just adding a system property is easier than creating/adding the servlet extension mechanism to their own deployment.

```suggestion
    private static final boolean DEFAULT_PRESERVE_PATH_OF_FORWARD = Boolean.getBoolean("io.undertow.servlet.preserve-path-of-forward");

    private boolean preservePathOnForward = DEFAULT_PRESERVE_PATH_OF_FORWARD;
```